onmt/translate/translator.py

```diff
@@ -812,7 +812,7 @@ def _translate_batch(self, batch, data):
             # Loop over the batch_size number of beam
             for j, b in enumerate(beam):
                 b.advance(out[j, :],
-                          beam_attn.data[j, :, :memory_lengths[j]])
+                          beam_attn.data[j, :, :int(memory_lengths[j])])
```

Comment:
why is this casting suddenly required ?

Reply:
Oops, accidentally made the ``lengths`` tensor ``self.dtype`` (float) in ``AudioSeqField`` and didn't think twice about casting a tensor to an integer. Should've been kind of obvious since it worked for text... Anyways, fixed.